### PR TITLE
Feature/contract info

### PR DESCRIPTION
**Pull request by LRUki**
displays contract constructs/messages on contract load in the modal. 
There is an issue regarding the useABi in ContractInfo.vue. (watch doesn't get triggered on uploading file)
some of the codes can be refactored once the issue is resolved
Screenshot 2021-06-14 at 04 34 48

### CODE REVIEW

**Comment by sirius651** [comment]
while debugging on this, I found out that `onChangeAbi` function to modulates the abi is not invoked. and It seems to work well in other uses. so, it should be work after `onChangeAbi` called.
And, could you pull on latest branch `main` with this branch?
@LRUki

**Comment by LRUki** [comment]
@sirius651 i still dont quite understand. do you mean we have to use onChangeAbi in the contractInfo.vue to observe the abi change? shouldn't `abi` itslef update properfly for later use anyway??

**Comment by hoonsubin** [comment]
Please fix the styles! I'll review this after it has been turned to ready for review!

**Comment by LRUki** [comment]
@hoonsubin  those style issues is stemming from the target branch: feature/deploy-contract, which @sirius651 is still working on. I'll have to fix the styles of the file will is still working on but would that be fine?

**Comment by hoonsubin** [comment]
Ah, in that case waiting would be better. Thanks for letting me know

**Comment by sirius651** [comment]
Merged and resolved conflicts based on 'feature/contract-info' @hoonsubin @LRUki

File: src/components/dapps/Message.vue [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      </span>
+    </div>
+  </div>
+</template>
+<script lang="ts">
+// import useAbi from '@/hooks/useAbi';
+import { PropType, defineComponent } from 'vue';
+import type { AbiParam } from '@polkadot/api-contract/types';
+import type { TypeDef } from '@polkadot/types/create/types';
+interface MessageType {
```
Comment (hoonsubin) [inline src/components/dapps/Message.vue]: The same type is being repeated on `ContractInfo` component https://github.com/PlasmNetwork/plasm-apps-vue/pull/28/files#diff-797a2b6e48f6a83f652ea849b20fbb8f19a7e18c4f9c18484d433520ab983474R16 and [here](https://github.com/PlasmNetwork/plasm-apps-vue/pull/28/files/20e17626fca2b0fa7cb15ea7b6c7acc751f50026#diff-68f0cafb24e9ce6767b2401349063096e4974e009d8c9f16e9b7389ac3e24ac2R476)
If we are using this in multiple places, please move this to a separate typescript file.

File: src/components/dapps/ModalCreateDapps.vue [hunk 2]
```diff
@@ -167,7 +167,31 @@
...
+                        :key="item"
+                        :value="item"
+                      >
+                        {{ item }}
+                      </option>
+                    </select> -->
+                </div>
+
+                <div>
+                  <label
```
Comment (hoonsubin) [inline src/components/dapps/ModalCreateDapps.vue]: I think this is an empty label

**Comment by sirius651** [comment]
@hoonsubin  I pushed the codes which improve UX for forgetting contract && resolve above issues. 
if no critical issue, we'd better merge now.

**Comment by hoonsubin** [comment]
Thank you for the addition. @LRUki can I merge this PR now?

**Comment by LRUki** [comment]
@hoonsubin havent changed anything so go ahead ! The refactor u mentioned in the comment I’ll fix it when useAbi is fixed

**Pull request merged**